nixos/tests/matrix/matrix-alertmanager.nix

```
@@ -0,0 +1,190 @@
+
+              listeners = [
+                {
+                  # The default but tls=false
+                  bind_addresses = [
+                    "0.0.0.0"
+                  ];
+                  port = 8448;
+                  resources = [
+                    {
+                      compress = true;
+                      names = [ "client" ];
+                    }
+                    {
+                      compress = false;
+                      names = [ "federation" ];
+                    }
+                  ];
+                  tls = false;
+                  type = "http";
+                  x_forwarded = false;
+                }
+              ];
+            };
+          };
+
+          networking.firewall.allowedTCPPorts = [ 8448 ];
+
+          environment.systemPackages = [
+            (pkgs.writeShellScriptBin "register_alertmanager_user" ''
+              exec ${pkgs.matrix-synapse}/bin/register_new_matrix_user \
+                -u alertmanager \
+                -p alertmanager-password \
+                --admin \
+                --shared-secret supersecret-registration \
+                http://localhost:8448
+            '')
+            # This is needed to solve a chicken and egg
+            # problem. Matrix-alertmanager expects a token for authentication,
+            # but a token is create after the user has been registered. This
+            # changes the token in the database to match the one specified in
+            # the service settings.
+            (pkgs.writers.writePython3Bin "hardcode_matrix_values"
+              {
+                libraries = with pkgs.python3Packages; [
+                  sqlite-utils
+                ];
+              }
+              ''
+                import sqlite3
+                con = sqlite3.connect("/var/lib/matrix-synapse/homeserver.db")
+                cur = con.cursor()
+                cur.execute("update access_tokens set token='faketoken' where id = 2")
+                con.commit()
+                con.close()
+              ''
+            )
+          ];
+        };
+
+      matrix_alertmanager =
+        { pkgs, ... }:
+        {
+          services.matrix-alertmanager = {
+            enable = true;
+            tokenFile = pkgs.writeText "token.txt" "faketoken";
+            secretFile = pkgs.writeText "secret.txt" "wontbeused";
+            homeserverUrl = "http://homeserver:8448";
+            matrixRooms = "matrix/!123:localhost:8448";
+            matrixUser = "alertmanager";
+          };
+        };
+
+      client =
+        { pkgs, ... }:
+        {
+          environment.systemPackages = [
+            (pkgs.writers.writePython3Bin "create_room_and_join"
+              {
+                libraries = with pkgs.python3Packages; [
+                  matrix-nio
+                ];
+              }
+              ''
+                import asyncio
+
+                from nio import (
+                    AsyncClient,
+                )
+
```

uber nit, feel free to ignore: formatting seems a little verbose here. Perhaps tighten it up?

```suggestion
                import asyncio
                from nio import AsyncClient
```

---

This is copy-pasted from the `mjolnir` module test, running the script against https://github.com/psf/black doesn't change this particular line so I'm not sure if there's another formatter that does this. Personally I strive to keep things as uniform to other modules as possible, so I'll skip this recommendation.

---

I completely removed this, as I realized a token is created by default when registering the user and there's no need to login the user manually.